pkg/controller/azure/database/mysqlservervirtualnetworkrule/managed_test.go

```diff
@@ -0,0 +1,434 @@
+}
+
+type virtualNetworkRuleModifier func(*v1alpha1.MysqlServerVirtualNetworkRule)
+
+func withConditions(c ...runtimev1alpha1.Condition) virtualNetworkRuleModifier {
+	return func(r *v1alpha1.MysqlServerVirtualNetworkRule) { r.Status.ConditionedStatus.Conditions = c }
+}
+
+func withType(s string) virtualNetworkRuleModifier {
+	return func(r *v1alpha1.MysqlServerVirtualNetworkRule) { r.Status.Type = s }
+}
+
+func withID(s string) virtualNetworkRuleModifier {
+	return func(r *v1alpha1.MysqlServerVirtualNetworkRule) { r.Status.ID = s }
+}
+
+func withState(s string) virtualNetworkRuleModifier {
+	return func(r *v1alpha1.MysqlServerVirtualNetworkRule) { r.Status.State = s }
+}
+
+func virtualNetworkRule(sm ...virtualNetworkRuleModifier) *v1alpha1.MysqlServerVirtualNetworkRule {
+	r := &v1alpha1.MysqlServerVirtualNetworkRule{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace:  namespace,
+			Name:       name,
+			UID:        uid,
+			Finalizers: []string{},
+		},
+		Spec: v1alpha1.VirtualNetworkRuleSpec{
+			ResourceSpec: runtimev1alpha1.ResourceSpec{
+				ProviderReference: &corev1.ObjectReference{Namespace: namespace, Name: providerName},
+			},
+			Name:              name,
+			ServerName:        serverName,
+			ResourceGroupName: resourceGroupName,
+			VirtualNetworkRuleProperties: v1alpha1.VirtualNetworkRuleProperties{
+				VirtualNetworkSubnetID:           vnetSubnetID,
+				IgnoreMissingVnetServiceEndpoint: true,
+			},
+		},
+		Status: v1alpha1.VirtualNetworkRuleStatus{},
+	}
+
+	for _, m := range sm {
+		m(r)
+	}
+
+	return r
+}
+
+// Test that our Reconciler implementation satisfies the Reconciler interface.
+var _ resource.ExternalClient = &external{}
+var _ resource.ExternalConnecter = &connecter{}
+
+func TestCreate(t *testing.T) {
+	cases := []testCase{
+		{
+			name: "SuccessfulCreate",
+			e: &external{client: &fake.MockMySQLVirtualNetworkRulesClient{
+				MockCreateOrUpdate: func(_ context.Context, _ string, _ string, _ string, _ mysql.VirtualNetworkRule) (mysql.VirtualNetworkRulesCreateOrUpdateFuture, error) {
+					return mysql.VirtualNetworkRulesCreateOrUpdateFuture{}, nil
+				},
+			}},
+			r: virtualNetworkRule(),
+			want: virtualNetworkRule(
+				withConditions(runtimev1alpha1.Creating()),
+			),
+		},
+		{
+			name: "FailedCreate",
+			e: &external{client: &fake.MockMySQLVirtualNetworkRulesClient{
+				MockCreateOrUpdate: func(_ context.Context, _ string, _ string, _ string, _ mysql.VirtualNetworkRule) (mysql.VirtualNetworkRulesCreateOrUpdateFuture, error) {
+					return mysql.VirtualNetworkRulesCreateOrUpdateFuture{}, errorBoom
+				},
+			}},
+			r: virtualNetworkRule(),
+			want: virtualNetworkRule(
+				withConditions(runtimev1alpha1.Creating()),
+			),
+			returnsErr: true,
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			_, err := tc.e.Create(ctx, tc.r)
+			if tc.returnsErr != (err != nil) {
+				t.Errorf("tc.e.Create(...) error: want: %t got: %t", tc.returnsErr, err != nil)
+			}
```

Comment:
Most other tests check that the expected error was returned with wrapped context. Is there a reason to avoid doing that here?

Reply:
I don't believe there is a specific reason to not do so. I implemented these tests following the same pattern as the only existing managed reconciler in the code base `controller/aws/cache/managed_test.go`, but I agree that errors should be checked. Updating in pending commit!